src/foam/u2/Tooltip.js

```diff
@@ -46,8 +43,65 @@ foam.CLASS({
   ],
 
   listeners: [
+    function onMouseOver() {
+      var timer;
+      var self = this;
+      this.target.on('mousemove', function(evt) {
+        if ( timer != undefined ) {
+          clearTimeout(timer);
+        }
+        timer = setTimeout(function() {
+          if ( self.opened === false ) {
+            self.document.body.insertAdjacentHTML('beforeend', self.outerHTML);
+
+            var domRect = self.target.el().getBoundingClientRect();
+            var screenWidth = self.window.innerWidth;
+            var screenHeight = self.window.innerHeight;
+            var scrollY = self.window.scrollY;
+            var selfRect = self.el().getBoundingClientRect();
+            var height = self.el().getBoundingClientRect().height;
+            var top = (domRect.top - scrollY > screenHeight / 2) ? evt.pageY - 30 - height : evt.pageY + 30;
+            var left = (domRect.left > screenWidth / 2) ? evt.pageX - 20 - selfRect.width  :  evt.pageX + 20;
+
+            self.opened = true;
+            self.load();
+            self.style({
+              'background': 'rgba(80, 80, 80, 0.9)',
+              'border-radius': '5px',
+              'color': 'white',
+              'padding': '5px 8px',
+              'position': 'absolute',
+              'z-index': '2000',
+              'max-width': (screenWidth / 4)+'px',
+              'top': top,
+              'left': left
+            });
+          }
+        }, 500);
+      });
+      
+      self.target.on('mousedown', function() {
```

Comment:
Better to make this function be a listener rather than repeating 4 times.

Reply:
Done